ydb/aio/query/pool.py

```diff
@@ -18,16 +20,67 @@
 logger = logging.getLogger(__name__)
 
 
-class QuerySessionPoolAsync:
-    """QuerySessionPoolAsync is an object to simplify operations with sessions of Query Service."""
+class QuerySessionPool:
+    """QuerySessionPool is an object to simplify operations with sessions of Query Service."""
 
-    def __init__(self, driver: common_utils.SupportedDriverType):
+    def __init__(self, driver: common_utils.SupportedDriverType, size: int = 100):
         """
         :param driver: A driver instance
+        :param size: Size of session pool
         """
 
-        logger.warning("QuerySessionPoolAsync is an experimental API, which could be changed.")
+        logger.warning("QuerySessionPool is an experimental API, which could be changed.")
         self._driver = driver
+        self._size = size
+        self._should_stop = asyncio.Event()
+        self._queue = asyncio.Queue()
+        self._current_size = 0
+        self._waiters = 0
+        self._loop = asyncio.get_running_loop()
+
+    async def _create_new_session(self):
+        session = QuerySession(self._driver)
+        await session.create()
+        logger.debug(f"New session was created for pool. Session id: {session._state.session_id}")
+        return session
+
+    async def acquire(self) -> QuerySession:
+        if self._should_stop.is_set():
+            logger.error("An attempt to take session from closed session pool.")
+            raise RuntimeError("An attempt to take session from closed session pool.")
+
+        session = None
+        try:
+            session = self._queue.get_nowait()
+        except asyncio.QueueEmpty:
+            pass
+
+        if session is None and self._current_size == self._size:
+            queue_get = asyncio.ensure_future(self._queue.get())
+            task_stop = asyncio.ensure_future(asyncio.ensure_future(self._should_stop.wait()))
+            done, _ = await asyncio.wait((queue_get, task_stop), return_when=asyncio.FIRST_COMPLETED)
+            if task_stop in done:
+                queue_get.cancel()
+                return await self._create_new_session()
```

Comment:
why it create a session on stop pool? 
may be runtime error - similar to acquire on stopped pool?